refactor: Article Uri 변경

yooooonshine opened the pull request:
## 개요
- close #35 

## 작업사항
- 변경된 API에 따라 ArticleController의  URI를 변경하였다.

yooooonshine committed:
refactor: Article Uri 변경

Lee-sh98 commented on hicc-reboot/src/main/java/hiccreboot/backend/controller/ArticleController.java:
```diff
@@ -10,113 +24,124 @@
+		@RequestParam(value = "size") int pageSize,
+		@RequestParam(value = "sort", required = false, defaultValue = "null") String sort,
+		@RequestParam(value = "search", required = false, defaultValue = "null") String search) {
+		//유저 이름,등급
+		Grade userGrade = Grade.APPLICANT;
+		String userName = "나야나";
+
```
이런 정보가 필요한 정보인가요??

Lee-sh98 commented on hicc-reboot/src/main/java/hiccreboot/backend/controller/ArticleController.java:
```diff
@@ -10,113 +24,124 @@
+	private final EntityManager em;
+
+	@GetMapping
+	public Object searchArticleList(
```
반환 타입도 확인부탁드립니다!

Lee-sh98 commented on hicc-reboot/src/main/java/hiccreboot/backend/controller/ArticleController.java:
```diff
@@ -10,113 +24,124 @@
+		String userName = "나야나";
+
+		if (sort.equals("null")) {
+			List<Article> articles = articleService.findArticles(pageNumber, pageSize).getContent();
```
controller에 if문에 따라 서비스 로직이 들어가는건 레이어 분리가 되지 않은것 같아요. 서비스 계층으로 내리는게 좋아보입니다.

Lee-sh98 commented on hicc-reboot/src/main/java/hiccreboot/backend/controller/ArticleController.java:
```diff
@@ -10,113 +24,124 @@
+		throw new ArticleNotFoundException();
+	}
+
+	private Object makeArticleDTO(Article article, Grade grade, String name) {
```
여기도 controller 클래스에 로직이 들어가는건 좋지 않아보입니다!

Lee-sh98 committed:
feat: Redis 기본 설정 (#43) (#44)

* docs: redis 의존성 주입

* feat: redis configuration

yooooonshine committed:
refactor: articleController에서 비즈니스 로직 분리

ArticleController에서 비즈니스로직부분은 Service계층으로 분리하였다.

yooooonshine committed:
refactor: ArticleList 조회 조건에 BoardType 추가

기존의 fidndArticlesByMemberName과
findArticlesBySubject, findArticles에 BoardType 조건을 추가하였다.

Lee-sh98 committed:
feat: Spring Security 적용 (#33) (#47)

* feat: token provider 인터페이스 구현

* feat: member repository 구현

* docs: 의존성 수정

* docs: git ignore 수정

* feat: JWT constant 추가

* feat: token provider 구현체

* feat: web security 생성

* feat: jwt authentication filter 생성

* feat: refresh token 엔티티 구현

* feat: refresh token repository 구현

* feat: Token provider 내 파라미터 변경, 불필요한 메소드 제거

* feat: Token provider 구현체 수정

- createRefreshToken 파라미터 변경
  - redis 적용에 따른 수정
  - issuer 수정
- disableRefreshToken 로직 변경
  - redis 적용에 따른 수정

* feat: access token 암호화 알고리즘 변경

* feat: jwt authentication filter 구현

* feat: user details service 구현체 추가

* feat: Grade description 수정

- GrantedAuthorityDefaults에 따른 변경

* feat: 학번, 비밀번호를 통한 인증 필터

* feat: 로그인 성공 핸들러 구현

* feat: 로그인 실패 핸들러 구현

* feat: RefreshToken 기본 생성자 추가

* feat: Redis 설정 변경

* feat: 토큰 반환을 위한 dto 생성

* feat: base response 내 timestamp 포맷팅

* feat: HttpServletResponse의 Writer 인스턴스 생성

* feat: 토큰 반환 방법을 body로 변경

* feat: ResponseWriter 적용

* refactor: 토큰 추출 로직 최적화

yooooonshine committed:
refactor: Article response에 대하여 Appendices에 대한 다형성 제거

response에 Appendices가 없을 경우 빈 리스트를 리턴하도록 하여, 다형성을 제거하였다.

yooooonshine committed:
feat: Article 관련 jwt, return 구현

yooooonshine committed:
chore: resolve conflict

yooooonshine committed:
chore: Member엔티티에 College 관련 제거

yooooonshine committed:
refactor: appendixExistence 제거

yooooonshine committed:
refactor: 연관관계 편의 메서드 생성자에서 처리하도록 변경

yooooonshine committed:
refactor: 연관관계 편의 메서드 생성자에서 처리하도록 변경

yooooonshine committed:
Merge remote-tracking branch 'origin/feature/35-refactor-article-uri' into feature/35-refactor-article-uri

yooooonshine committed:
chore: 매핑 정보 오타 수정

yooooonshine committed:
chore: ArticleRequest에 Getter 제거

yooooonshine committed:
refactor: transaction 추가
